include/open62541pp/types/Variant.h

```diff
@@ -483,7 +495,7 @@ void Variant::setArrayCopyImpl(InputIt first, InputIt last, const UA_DataType& d
     using ValueType = typename std::iterator_traits<InputIt>::value_type;
     const size_t size = std::distance(first, last);
     auto native = detail::allocateArrayUniquePtr<ValueType>(size, dataType);
-    std::transform(first, last, native.get(), [&](auto&& value) {
+    std::transform(first, last, native.get(), [&](const ValueType& value) {
```

Comment:
Instead of casting to ValueType, I just changed the type directly. 

If this is changed back to `auto&&`, the added test will fail.

Reply:
Best solution! 👍